sources/osg/State.js

```diff
@@ -285,111 +287,16 @@ State.prototype = MACROUTILS.objectLibraryClass( MACROUTILS.objectInherit( Objec
                 mu = this.modelViewNormalMatrix;
                 mul = uniformCache.uModelViewNormalMatrix;
                 if ( mul ) {
-
-                    normal[ 0 ] = matrix[ 0 ];
-                    normal[ 1 ] = matrix[ 1 ];
-                    normal[ 2 ] = matrix[ 2 ];
-                    normal[ 4 ] = matrix[ 4 ];
-                    normal[ 5 ] = matrix[ 5 ];
-                    normal[ 6 ] = matrix[ 6 ];
-                    normal[ 8 ] = matrix[ 8 ];
-                    normal[ 9 ] = matrix[ 9 ];
-                    normal[ 10 ] = matrix[ 10 ];
-
-                    mat4.invert( normal, normal );
-                    mat4.transpose( normal, normal );
-
-                    mu.setMatrix4( normal );
+                    mat3.normalFromMat4( normal, matrix );
+                    mu.setMatrix3( normal );
                     mu.apply( gc, mul );
                 }
-            }
-
-            this._modelViewMatrix = matrix;
```

Comment:
Hmm I don't know, maybe it's in the case applyModelViewMatrix is called several time in a row?